src/main/java/org/opentest4j/fixtures/FixtureManager.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j.fixtures;
+
+import static org.opentest4j.fixtures.Utils.checkNotNull;
+import static org.opentest4j.fixtures.Utils.rethrowAsException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * Manages the interaction of a {@link TestFixture}
+ *
+ * @author Kevin Cooney
+ * @since 1.1
+ */
+public final class FixtureManager {
+	private final Stack<TearDown> tearDowns = new Stack<TearDown>();
```

Comment:
`Deque` and `ArrayDeque` are more performant as a stack than `Stack`, according to https://docs.oracle.com/javase/6/docs/api/java/util/ArrayDeque.html.